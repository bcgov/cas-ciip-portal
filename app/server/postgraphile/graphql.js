const {
  createPostGraphileSchema,
  withPostGraphileContext,
} = require("postgraphile");
const { graphql } = require("graphql");
const authenticationPgSettings = require("./authenticationPgSettings");
const { pgPool, getDatabaseUrl } = require("../db/databaseConnectionService");
const postgraphileOptions = require("./postgraphileOptions");

const pgSettings = (req) => {
  const opts = {
    ...authenticationPgSettings(req),
  };
  return opts;
};

let postgraphileSchemaSingleton;

const postgraphileSchema = async () => {
  if (!postgraphileSchemaSingleton) {
    postgraphileSchemaSingleton = await createPostGraphileSchema(
      getDatabaseUrl(),
      process.env.DATABASE_SCHEMA || "ggircs_portal",
      postgraphileOptions()
    );
  }

  return postgraphileSchemaSingleton;
};

async function performQuery(query, variables, request) {
  const settings = pgSettings(request);

  return withPostGraphileContext(
    {
      pgPool,
      pgSettings: settings,
    },
    async (context) => {
      // `context` object, which should NOT be used outside of this
      // function.
      const result = graphql(
        await postgraphileSchema(),
        query,
        null,
        { ...context }, // You can add more to context if you like
        variables
      );
      return result;
    }
  );
}

module.exports = {
  performQuery,
};
