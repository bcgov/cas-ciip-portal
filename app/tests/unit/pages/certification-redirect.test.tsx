import React from 'react';
import {shallow} from 'enzyme';
import CertificationRedirect from 'pages/certifier/certification-redirect';
import {certificationRedirectQueryResponse} from 'certificationRedirectQuery.graphql';

const query: certificationRedirectQueryResponse['query'] = {
  session: {
    ciipUserBySub: {
      id: 'sjkdfh'
    },
    ' $fragmentRefs': {
      defaultLayout_session: true
    }
  },
  certificationUrlByRowId: {
    id: 'sjkdhf',
    rowId: 'lsdjf',
    expiresAt: '2120-06-02 11:20:24.934381-07',
    ciipUserByCreatedBy: {
      firstName: 'Old',
      lastName: 'MacDonald'
    },
    applicationByApplicationId: {
      id: 'kdfbo',
      latestDraftRevision: {
        versionNumber: 1
      },
      facilityByFacilityId: {
        facilityName: 'Farm',
        organisationByOrganisationId: {
          operatorName: "MacDonald's Agriculture Ltd."
        }
      }
    }
  }
};

describe('Login redirect page', () => {
  it('matches the last accepted Snapshot', () => {
    const wrapper = shallow(
      <CertificationRedirect query={query} router={null} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
