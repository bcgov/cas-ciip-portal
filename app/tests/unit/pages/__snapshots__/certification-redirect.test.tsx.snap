// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login redirect page matches the last accepted Snapshot 1`] = `
<Relay(DefaultLayout)
  session={
    Object {
      " $fragmentRefs": Object {
        "defaultLayout_session": true,
      },
      "ciipUserBySub": Object {
        "id": "sjkdfh",
      },
    }
  }
  title="Submission Certification"
>
  <Row
    noGutters={false}
  >
    <Col
      md={
        Object {
          "offset": 3,
          "span": 6,
        }
      }
    >
      <h2
        className="blue"
      >
        Your certification is requested.
      </h2>
      <p>
        <strong>
          Old MacDonald
        </strong>
         on behalf of
         
        <strong>
          MacDonald's Agriculture Ltd.
        </strong>
        , for facility
         
        <strong>
          Farm
        </strong>
         has requested that you review, certify, and approve the information contained in this CIIP application.
      </p>
      <p>
        Please continue to the certification page.
      </p>
      <Col
        md={
          Object {
            "offset": 1,
            "span": 5,
          }
        }
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="primary"
        >
          Continue
        </Button>
      </Col>
    </Col>
  </Row>
  <style>
    
          h2 {
            margin-bottom: 20px;
          }
        
  </style>
</Relay(DefaultLayout)>
`;
